lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/test/test.main.js

```diff
@@ -128,6 +128,94 @@ tape( 'the function computes the cumulative sum', function test( t ) {
 	t.end();
 });
 
+tape( 'the function computes the cumulative sum (accessors)', function test( t ) {
+	var expected;
+	var x;
+	var y;
+	var i;
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	y = [ 0.0, 0.0, 0.0, 0.0, 0.0 ];
+
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+	expected = [
+		1.0,
+		3.0,
+		6.0,
+		10.0,
+		15.0
+	];
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	y = [ 0.0, 0.0, 0.0, 0.0, 0.0 ];
+
+	gcusumkbn2( x.length, 10.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+	expected = [
+		11.0,
+		13.0,
+		16.0,
+		20.0,
+		25.0
+	];
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	x = [ NaN, NaN ];
+	y = [ 0.0, 0.0 ];
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+
+	for ( i = 0; i < y.length; i++ ) {
+		t.strictEqual( isnan( y[ i ] ), true, 'returns expected value. i: ' + i );
+	}
+
+	x = [ 1.0, NaN, 3.0, NaN ];
+	y = [ 0.0, 0.0, 0.0, 0.0 ];
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+
+	expected = [
+		1.0,
+		NaN,
+		NaN,
+		NaN
+	];
+	for ( i = 0; i < y.length; i++ ) {
+		if ( isnan( expected[ i ] ) ) {
+			t.strictEqual( isnan( y[ i ] ), true, 'returns expected value. i: ' + i );
+		} else {
+			t.strictEqual( y[ i ], expected[ i ], true, 'returns expected value. i: ' + i );
+		}
+	}
+
+	x = [ 1.0, 1.0e100, 1.0, -1.0e100 ];
+	y = [ 0.0, 0.0, 0.0, 0.0 ];
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+
+	expected = [
+		1.0,
+		1.0e100,
+		1.0e100,
+		2.0
+	];
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	x = [];
+	y = [];
+	expected = [];
+	for ( i = 0; i < 1e3; i++ ) {
+		x.push( i+1 );
+		y.push( 0.0 );
+		if ( i === 0 ) {
+			expected.push( x[ i ] );
+		} else {
+			expected.push( expected[ i-1 ] + x[ i ] );
+		}
+	}
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, toAccessorArray( y ), 1 );
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	t.end();
+});
+
 tape( 'the function returns a reference to the output array', function test( t ) {
 	var out;
 	var x;
```

Comment:
@headlessNode I suggest adding a test case here when providing accessor arrays. This would have caught the bug mentioned above.